src/App.tsx

```diff
@@ -1,16 +1,26 @@
+import { useState } from 'react';
 import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import { NewMovie } from './components/NewMovie';
 import moviesFromServer from './api/movies.json';
+import { Movie } from './types/Movie';
 
 export const App = () => {
+  const [movies, setMovies] = useState<Movie[]>(moviesFromServer);
+
+  const onAdd = (movie: Movie) => {
+    setMovies(currentMovie => (
+      [...currentMovie, movie]
+    ));
+  };
+
   return (
     <div className="page">
       <div className="page-content">
-        <MoviesList movies={moviesFromServer} />
+        <MoviesList movies={movies} />
       </div>
       <div className="sidebar">
-        <NewMovie /* onAdd={(movie) => {}} */ />
+        <NewMovie onAdd={onAdd} />
```

Comment:
```suggestion
        <NewMovie onAdd={onAdd} key={count} />
```
Add key counter here to force re-render component